test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildServerIntegrationTest.cs

```diff
@@ -103,5 +103,35 @@ public Task Build_SimpleMvc_WithServer_UsingDesktopMSBuild_CanBuildSuccessfully(
             Assert.FileExists(result, IntermediateOutputPath, "SimpleMvc.dll");
             Assert.FileDoesNotExist(result, IntermediateOutputPath, "SimpleMvc.Views.dll");
         }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task Build_ServerConnectionMutexCreationFails_FallsBackToInProcessRzc()
+        {
+            // Use a pipe name longer that 260 characters to make the Mutex constructor throw.
```

Comment:
Mutex constructor would throw if the mutex name is longer than 260 characters. Since we use the pipe name as the base for mutex name, I found it easier to use this to test this case.

Reply:
Nice